File: BI-TJV/semestral/rest-server/src/main/java/wrzecond/controller/TjvOrderController.java
package wrzecond.controller;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;
import wrzecond.order.TjvOrderCreateDTO;
import wrzecond.order.TjvOrderReadDTO;
import wrzecond.entity.TjvOrder;
import wrzecond.service.TjvEmployeeService;
import wrzecond.service.TjvOrderService;

import javax.servlet.http.HttpServletRequest;
import java.util.List;

@RestController
@TjvController.Visibility (
    create = TjvController.VisibilitySettings.LOGGED,
    findAll = TjvController.VisibilitySettings.LOGGED,
    update = TjvController.VisibilitySettings.LOGGED,
    delete = TjvController.VisibilitySettings.LOGGED,
    getByID = TjvController.VisibilitySettings.NONE
)
@RequestMapping("/orders/")
public class TjvOrderController extends TjvController<TjvOrder, TjvOrderReadDTO, TjvOrderCreateDTO> {

    private final TjvOrderService orderService;

    @Autowired
    public TjvOrderController (TjvOrderService orderService, TjvEmployeeService authService) {
        super (orderService, authService);
        this.orderService = orderService;
    }

    @GetMapping("at/{id}")
    List<TjvOrderReadDTO> findByTableID (@PathVariable int id, HttpServletRequest request) {
        authenticate(request, VisibilitySettings.LOGGED);
        return orderService.findByTableID(id);
    }

    @Override
    protected String getCreateLocationPath () {
        return "/orders/%d";
    }

}

File: BI-KOT/semestral/rest-server/src/main/kotlin/wrzecond/controller/TjvOrderController.kt
package wrzecond.controller

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import wrzecond.TjvOrderCreateDTO
import wrzecond.TjvOrderReadDTO
import wrzecond.TjvOrderUpdateDTO
import wrzecond.entity.TjvOrder
import wrzecond.service.TjvEmployeeService
import wrzecond.service.TjvOrderService
import javax.servlet.http.HttpServletRequest

@RestController
@RequestMapping("/orders/")
@TjvController.Visibility (
    create = TjvController.VisibilitySettings.LOGGED,
    findAll = TjvController.VisibilitySettings.LOGGED,
    update = TjvController.VisibilitySettings.LOGGED,
    delete = TjvController.VisibilitySettings.LOGGED,
    getByID = TjvController.VisibilitySettings.NONE
)
class TjvOrderController (override val service: TjvOrderService, employeeService: TjvEmployeeService)
: TjvController<TjvOrder, TjvOrderReadDTO, TjvOrderCreateDTO, TjvOrderUpdateDTO> (service, employeeService) {

    override val createLocationPath: String
        get() = "/orders/%d"

    @GetMapping("at/{id}")
    fun findByTableID (@PathVariable id: Int, request: HttpServletRequest)
        = authenticate(request, VisibilitySettings.LOGGED) { service.findByTableID(id) }

}